### PR TITLE
RARO-629: Implement Export-to-csv Feature to journals' Advanced Search capability

### PR DESCRIPTION
**Pull request by sjtower**

### CODE REVIEW

File: src/main/java/org/ambraproject/wombat/service/remote/ArticleSearchQuery.java [hunk 1]
```diff
@@ -51,6 +51,12 @@
   private static final String RSS_FIELDS = Joiner.on(',').join(ImmutableList.copyOf(new String[]{
       "id", "publication_date", "title", "title_display", "journal_name", "author_display",
       "abstract", "abstract_primary_display"}));
+  private static final String CSV_FIELDS = Joiner.on(',').join(ImmutableList.copyOf(new String[]{
```
Comment (randy-tano) [inline src/main/java/org/ambraproject/wombat/service/remote/ArticleSearchQuery.java]: I think `Joiner.join()` will take a list of objects, so you simplify it to:

```
CSV_FIELDS = Joiner.on(',').join("id", "publication_date", "title",
      "author_display", "author_affiliate", "article_type",
      "received_date", "accepted_date", "counter_total_all", "alm_scopusCiteCount",
      "alm_citeulikeCount", "alm_connoteaCount", "alm_mendeleyCount", "alm_twitterCount",
      "alm_facebookCount", "alm_pmc_usage_total_all", "alm_webOfScienceCount", "editor_display",
      "abstract", "subject", "reference");
```

File: src/main/java/org/ambraproject/wombat/controller/SearchController.java [hunk 2]
```diff
@@ -636,13 +645,61 @@ private ModelAndView getFeedModelAndView(Site site, String feedType, String titl
...
+    final String filename = String.format("solrCsvExport-%s-q-%s.csv", Instant.now(),
+        params.getFirst("q"));
+    response.setHeader("Content-Disposition", "attachment; filename=" + filename);
+    return convertToCsvFile(collateCsvResults(request, model, site, params, totalRows));
+  }
+
+  private String collateCsvResults(HttpServletRequest request, Model model, Site site,
+                                   MultiValueMap<String, String> params, Integer totalRows) throws IOException {
+    StringBuilder resultsBuilder = new StringBuilder();
+    Integer start = 0;
```
[the remark this answers is not in the document]
Comment (sjtower) [reply src/main/java/org/ambraproject/wombat/controller/SearchController.java]: yes, start gets incremented by 1000 if requesting over 1000 rows.

**Comment by sjtower** [comment]
Implemented code review changes, please re-review @randy-tano

File: src/main/java/org/ambraproject/wombat/controller/SearchController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sjtower) [reply src/main/java/org/ambraproject/wombat/controller/SearchController.java]: Sorry I misspoke. We don't need 'start' at all - we can just use `i`

**Pull request merged**